### PR TITLE
Improved installation instructions for OpenShift

### PR DESCRIPTION
**Pull request by makuhama**
Instructions for creating the pull secret have been fixed.

Removed hard-wired namespace 'mongodb' in resources defined in enterprise-mongodb-openshift.yaml

Instead of instructing the reader to edit enterprise-mongodb-openshift.yaml the service accounts defined there are now linked to a secret named 'openshift-pull-secret'. This in sync with the instructions in 'openshift-install.md' and significantly reduces the amount of OpenShiftf specific instructions.

### CODE REVIEW

**Comment by BenElgar** [comment]
Thank you for your contribution! We're reviewing this internally and tracking this as [CLOUDP-55732](https://jira.mongodb.org/browse/CLOUDP-55732) (private link).

**Comment by makuhama** [comment]
Thanks for taking care of my PR. Just in case it matters: I have no access at all to the above mentioned JIRA ticket (not even to the project CLOUDP). Just signed up to MongoDB's single signon service. You should be able to find my account via my real life name.

**Comment by jamesbroadhead** [comment]
Hi @makuhama - 

Apologies for the delay in reviewing this one. 

Could you please sign our CLA, so that we can merge it? 

https://www.mongodb.com/legal/contributor-agreement

all the best, 

James

**Comment by makuhama** [comment]
@jamesbroadhead CLA signed

**Comment by jamesbroadhead** [comment]
@makuhama  thank you!

I've asked @rodrigovalin to do a review pass over the PR

**Comment by makuhama** [comment]
merged from upstream/master
fixed conflicts
removed service account mongodb-enterprise-database-pods as referrenced nowhere (not even in https://github.com/mongodb/mongodb-kubernetes-operator)

File: helm_chart/templates/database-roles.yaml [hunk 1]
```diff
@@ -9,17 +9,6 @@ metadata:
   namespace: {{ .Values.namespace }}
   {{- end }}
 
----
```
Comment (antonlisovenko) [inline helm_chart/templates/database-roles.yaml]: This service account is necessary for database pods created by the Operator

File: mongodb-enterprise-openshift.yaml [hunk 2]
```diff
@@ -54,39 +53,37 @@ rules:
   - delete
   - update
 
-# This ClusterRoleBinding is necessary in order to use validating
+# This RoleBinding is necessary in order to use validating
 # webhooks—these will prevent you from applying a variety of invalid resource
 # definitions. The validating webhooks are optional so this can be removed if
 # necessary.
 ---
-kind: ClusterRoleBinding
```
Comment (antonlisovenko) [inline mongodb-enterprise-openshift.yaml]: what's the reason for this change?

File: mongodb-enterprise-openshift.yaml [hunk 3]
```diff
@@ -106,20 +103,19 @@ rules:
   - list
   - watch
 
+
 ---
-kind: ClusterRoleBinding
```
Comment (antonlisovenko) [inline mongodb-enterprise-openshift.yaml]: what's the reason for this change? CSR is a cluster resource, my understanding is that to grant permissions to it we need `ClusterRoleBinding`?

File: mongodb-enterprise-openshift.yaml [hunk 4]
```diff
@@ -152,20 +142,41 @@ rules:
...
   kind: Role
   name: mongodb-enterprise-appdb
 subjects:
   - kind: ServiceAccount
     name: mongodb-enterprise-appdb
-    namespace: mongodb
+
+
+---
+# Source: mongodb-enterprise-operator/templates/serviceaccount.yaml
```
Comment (antonlisovenko) [inline mongodb-enterprise-openshift.yaml]: Good idea with `imagePullSecrets` for service accounts
though the `mongodb-enterprise-openshift.yaml` already contains instructions for these two service accounts

**Comment by rodrigovalin** [comment]
@makuhama There are a few unresolved comments that have been there for a few months now.

I'm going to close this PR. Please re-open if you think you'll have the time to address those.

Thanks again for contributing to the Operator.
